### PR TITLE
feature/sc-32771/learningobject-bundling-module-route-updates

### PR DESCRIPTION
**Pull request by joaquinVill**
Wanted to request to see if I am on the right track with altering the bundling routes. Not a final pr. Thank you!

### CODE REVIEW

**Review by stef359 — CHANGES_REQUESTED** [review]
Change the target branch from 'main' to 'clark-service-refactor'

File: src/app/core/learning-object-module/bundling/bundling.service.ts [hunk 1]
```diff
@@ -39,4 +50,49 @@ export class BundlingService {
...
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
+  }
+
+
+  downloadObject(event: MouseEvent, object: LearningObject, index: number) {
+    event.stopPropagation();
+    this.currentIndex = index;
+    this.downloading[index] = true;
+    this.showDownloadModal = true;
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(object.id));
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/bundling/bundling.service.ts:93]: The downloadBundle function in libraryService should be moved to the bundling.service.ts file.

File: src/app/cube/details/components/action-panel/action-panel.component.ts [hunk 2]
```diff
@@ -199,10 +203,10 @@ export class ActionPanelComponent implements OnInit, OnDestroy {
...
    */
 
-  download(learningObjectId: string) {
-    this.toggleDownloadModal(true);
-    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
-  }
+  // download(learningObjectId: string) {
+  //   this.toggleDownloadModal(true);
+  //   this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
+  // }
```
Comment (Cwagne17) [inline src/app/cube/details/components/action-panel/action-panel.component.ts:209]: We need this. And this.libraryService should be this.bundlingService after the migration to bundling.service.ts.

File: src/app/cube/details/components/action-panel/action-panel.component.ts [hunk 3]
```diff
@@ -227,9 +231,9 @@ export class ActionPanelComponent implements OnInit, OnDestroy {
     this.toaster.success('Success!', 'Attribution information has been copied to your clipboard!');
   }
 
-  toggleDownloadModal(val?: boolean) {
-    this.showDownloadModal = val;
-  }
+  // toggleDownloadModal(val?: boolean) {
+  //   this.showDownloadModal = val;
+  // }
```
Comment (Cwagne17) [inline src/app/cube/details/components/action-panel/action-panel.component.ts:236]: We need this

File: src/app/cube/library/library.component.ts [hunk 4]
```diff
@@ -225,22 +227,22 @@ export class LibraryComponent implements OnInit, OnDestroy {
...
-    this.downloading[index] = false;
-  }
+  // downloadObject(event: MouseEvent, object: LearningObject, index: number) {
+  //   event.stopPropagation();
+  //   this.currentIndex = index;
+  //   this.downloading[index] = true;
+  //   this.showDownloadModal = true;
+  //   this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(object.id));
+  //   this.downloading[index] = false;
+  // }
```
Comment (Cwagne17) [inline src/app/cube/library/library.component.ts:245]: We need this.

File: src/app/core/library-module/library.service.ts [hunk 5]
```diff
@@ -133,48 +138,6 @@ export class LibraryService {
       .toPromise();
   }
 
-  /**
-   * Service function to download a learning object bundle.
-   * The call to download the bundle is made in @function downloadBundle()
-   *
-   * @param learningObjectId the mongo id of the learning object
-   */
-  learningObjectBundle(learningObjectId: string) {
```
Comment (Cwagne17) [inline src/app/core/library-module/library.service.ts:142]: Was this function not being used? If so, where was it moved to?

File: src/app/core/library-module/library.service.ts [hunk 6]
```diff
@@ -133,48 +138,6 @@ export class LibraryService {
...
-      (error) => {
-        this._loading$.next(false);
-        this.toaster.error('Preparation failed', error.message);
-      }
-    );
-  }
-
   /**
    * Method to start bundle stream and download the zip file
    * @param url request to api for zip in stream
```
Comment (Cwagne17) [inline src/app/core/library-module/library.service.ts:143]: This downloadBundle function just below here should be moved to bundling.service.ts

File: src/app/core/learning-object-module/bundling/bundling.service.ts [hunk 7]
```diff
@@ -39,4 +50,49 @@ export class BundlingService {
...
+    this.showDownloadModal = val;
+  }
+
+  download(learningObjectId: string) {
+    this.toggleDownloadModal(true);
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
+  }
+
+
+  downloadObject(event: MouseEvent, object: LearningObject, index: number) {
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/bundling/bundling.service.ts:88]: This looks like the function from the component, this should stay in the component and the reference to this.libraryService.downloadBundle should be this.bundlingService.downloadBundle and the downloadBundle function from library.service.ts should be moved here, then this should work.

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

**Comment by Cwagne17** [comment]
> Wanted to request to see if I am on the right track with altering the bundling routes. Not a final pr. Thank you!

make sure on the next story to make the branch name the shortcut branch name.

File: src/app/core/learning-object-module/bundling/bundling.service.ts (hunk 1 above)
Comment (pzalep1) [reply src/app/core/learning-object-module/bundling/bundling.service.ts:93]: +1

File: src/app/cube/details/components/action-panel/action-panel.component.ts (hunk 3 above)
Comment (pzalep1) [reply src/app/cube/details/components/action-panel/action-panel.component.ts:236]: +1

File: src/app/core/learning-object-module/bundling/bundling.service.ts [hunk 8]
```diff
@@ -39,4 +50,49 @@ export class BundlingService {
       return throwError(error);
     }
   }
+
+  toggleBundle(
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/bundling/bundling.service.ts:54]: I just checked in clark-service TOGGLE_BUNDLE_FILE is actually supposed to be in the file service but we didn't move it over in https://app.shortcut.com/clarkcan/story/32778/learningobject-file-module-route-updates-for-clark-refactor 
These are the only routes that should be in bundling service. and they map to this file in clark-service. https://github.com/Cyber4All/clark-service/blob/main/src/modules/learning-object-module/bundling/bundling.routehandler.ts
Screenshot 2024-09-16 at 10 27 09 AM

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**